pkg/apis/ocs/v1/storagecluster_types.go

```diff
@@ -25,16 +25,34 @@ type StorageClusterSpec struct {
 }
 
 // StorageDeviceSet defines a set of storage devices.
-// It is derived from and mapped to StorageClassDeviceSet in Rook.
+// It configures the StorageClassDeviceSets field in Rook-Ceph.
 type StorageDeviceSet struct {
-	Name string `json:"name"`
+	// Count is the number of devices in each StorageClassDeviceSet
 	// +kubebuilder:validation:Minimum=1
-	Count           int                          `json:"count"`
-	Resources       corev1.ResourceRequirements  `json:"resources"`
-	Placement       rookalpha.Placement          `json:"placement"`
+	Count int `json:"count"`
+
+	// Replica is the number of StorageClassDeviceSets for this
+	// StorageDeviceSet
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	Replica int `json:"replica,omitempty"`
```

Comment:
Can we start using this new `Replica` option in the default test StorageCluster that our functional tests deploy and test with?

That can be found here, ```pkg/deploy-manager/storagecluster.go``` in the ```defaultStorageCluster()``` function. Making this change will ensure we can deploy using a config that more closely resembles what we're going with for production. 

You might also consider updating the ```deploy/crds/ocs_v1_storagecluster_cr.yaml``` example CR.

Reply:
I'd like to keep the default test StorageCluster reflecting the current state of the product. Since the OCP console is currently stuck using a Count of 3 and a Replica of 0 (default 3), we should keep it that way until the console is updated.

Reply:
Somehow we need to functionally prove this configuration works. Otherwise we run the risk of breaking it. 

do we need to restructure the functional tests to allow us to cycle through multiple StorageCluster configurations?

Reply:
Simply testing that the expected number of OSDs gets created should suffice for now. I don't think we should spend time testing multiple StorageCluster configurations since we will only support one for the initial release.